autogluon/utils/tabular/features/generators/bulk.py

```diff
@@ -0,0 +1,155 @@
+import logging
+from typing import List
+
+import pandas as pd
+from pandas import DataFrame
+
+from .abstract import AbstractFeatureGenerator
+from ..feature_metadata import FeatureMetadata
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: Add parameter to add prefix to each generator to guarantee no name collisions: 'G1_', 'G2_', etc.
+# TODO: Add argument keep_unused, which creates an identity feature generator at each stage to pipe unused input features into the next stage instead of dropping them.
+class BulkFeatureGenerator(AbstractFeatureGenerator):
+    """
+    BulkFeatureGenerator is used for complex feature generation pipelines where multiple generators are required, with some generators requiring the output of other generators as input (multi-stage generation).
+    For ML problems, it is expected that the user uses a feature generator that is an instance of or is inheriting from BulkFeatureGenerator, as single feature generators typically will not satisfy the feature generation needs of all input data types.
```

Comment:
Add:  "Unless you are an expert user, we recommend you create custom FeatureGenerators based off of PipelineFeatureGenerator instead of BulkFeatureGenerator."

Reply:
Added